identity-provider-service/src/bin/identity_verifier.rs

```diff
@@ -129,28 +126,49 @@ async fn main() {
                                 .body("id_cred_pub not present.".to_string());
                         }
                     };
-                    let id_cred_pub_hash_bytes = hex::decode(&id_cred_pub_hash).unwrap();
-                    input.remove("id_cred_pub");
 
-                    let id_cred_pub_signature = match input.get("id_cred_pub_signature") {
-                        Some(sig) => sig.clone(),
+                    let endpoint_version = match input.get("endpoint_version") {
+                        Some(version) => version.clone(),
                         None => {
                             return Response::builder()
                                 .status(StatusCode::BAD_REQUEST)
-                                .body("id_cred_pub_signature not present.".to_string());
+                                .body("endpoint_version not present.".to_string());
                         }
                     };
-                    input.remove("id_cred_pub_signature");
+                    input.remove("endpoint_version");
 
-                    let endpoint_version = match input.get("endpoint_version") {
-                        Some(version) => version.clone(),
+                    if input.get("fail").is_some() {
+                        let delay = input
+                            .get("fail_delay")
+                            .cloned()
+                            .unwrap_or_else(|| "10".to_string());
```

Comment:
Shouldn't fail_delay always be there if `fail` is?

In that case I think it's better to reject instead of unwrap, no?